### PR TITLE
Add Nebius AI Studio LLM and embedding providers

### PR DESCRIPTION
**Pull request by vadjs**
## Description

Added Nebius AI Studio LLM and embedding provider.

## Checklist

- [x] The base branch of this PR is `dev`, rather than `main`
- [x] The relevant docs, if any, have been updated or created

## Screenshots

No visual changes

## Testing

It's not a feature, it's a new provider. The testing is simple: User Nebius provider with their API key. It just works.

### CODE REVIEW

File: core/llm/llms/Nebius.ts
Comment (RomneyDa) [inline core/llm/llms/Nebius.ts:1]: Hi @vadjs, nice addition, looks like Nebius is OpenAI compatible, could you switch this out for a simpler class that extends OpenAI? 

Kind of like:
https://github.com/continuedev/continue/blob/dev/core/llm/llms/Kindo.ts

or did I miss a reason for extending BaseLLM?

**Review by RomneyDa — CHANGES_REQUESTED** [review]

**Review by sestinj — CHANGES_REQUESTED** [review]
Thanks for the PR! I'd like to second the comment from Dallin: since it appears Nebius is OpenAI-compatible, I would prefer that this is just a subclass of the OpenAI class. I also noticed that the default base URL for the embeddings provider was a Voyage AI URL, I think this needs to be cleaned up

File: core/llm/llms/Nebius.ts
Comment (vadjs) [reply core/llm/llms/Nebius.ts:1]: Hi @RomneyDa, thank you for your advise!
I've changed `Nebius` provider accordingly. Now it extends `OpenAI`.

**Comment by vadjs** [comment]
Hi @sestinj,
Thank you for the good point. I've changed apiBase URL in the `NebiusEmbeddingsProvider` to the correct one.
Also extended `Nebius` LLM provider from `OpenAI` class as proposed before.

**Review by sestinj — APPROVED** [review]

**Comment by sestinj** [comment]
Awesome, thank you!

**Pull request merged**